File: CB - C/diagonal_negativos.c
#include <stdio.h>

int main()
{
    int n;

    printf("Qual ordem da matriz? ");
    scanf("%d", &n);

    int mat [n] [n];

    for (int i = 0; i < n; i++){
        for (int j = 0; j < n; j++){
            printf("Elemento [%d,%d]: ", i, j);
            scanf("%d", &mat[i] [j]);
        }
    }

      printf("DIAGONAL PRINCIAL:\n");
      for (int i = 0; i < n; i++){
            printf("%d ", mat[i] [i]);
        }


      int cont = 0;
      for (int i = 0; i < n; i++){
        for (int j = 0; j < n; j++){
                if (mat[i] [j] < 0){
                   cont++;
                }
        }
    }

    printf("\n");
    printf("Quantidade de negativos: %d", cont);



    return 0;
}

File: ws - eclipse/diagonal_negativos.java
import java.util.*;

public class diagonal_negativos {

	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		
		System.out.print("Qual a ordem da matriz? ");
		int n = sc.nextInt();
		
		double[][] mat = new double [n][n];
		
		for(int i = 0; i < n; i++) {
			for(int j = 0; j < n; j++) {
				System.out.print("ELEMENTO: [" + i + "," + j + "]: ");
				mat[i][j] = sc.nextDouble();
			}
		}
		
		System.out.println("DIAGONAL PRINCIPAL: ");
		for(int i = 0; i < n; i++) {
			System.out.print(mat[i][i] + " ");
		}
		
		int quanti = 0;
		for(int i = 0; i < n; i++) {
			for(int j = 0; j < n; j++) {
				if(mat[i][j] < 0) {
					quanti = quanti + 1;
				}
			}
		}
		System.out.print("\nQUANTIDADE DE NEGATIVOS = " + quanti);
		
		
		sc.close();
	}

}
